### ISSUE TITLE
Question about folder name strategy "a priori" ?

### ISSUE THREAD

**Issue by reyman** [papis/papis#359]
Hi !

I see that https://github.com/papis/papis/issues/295 to rename folder a posteriori.
Is there an option to configure the name of folder "a priori", for example, creating folder using this policy/strategy : `{author}-{date}-{uid}` ?
Best

**Comment by alejandrogallo** [comment]
do you mean like someting like this:
https://papis.readthedocs.io/en/latest/configuration.html#config-settings-add-folder-name
? Or more special?
Cheers!

**Comment by reyman** [comment]
Oh right, didn't see that because my base was imported from zotero ! 
Great to see there is a config option !
Thx !

**Comment by reyman** [comment]
Hum, is there a way to get the hash from doc object ? 

```
docs = papis.api.get_all_documents_in_lib(library_name)
for doc in docs:
    print(doc)
```

Dictionary returned by this function doesn't contain the =path= with =hash= and/or the =hash=
Any idea @alejandrogallo ?

**Comment by alejandrogallo** [comment]
This is the hashing function being used in `add` command
https://github.com/papis/papis/blob/master/papis/commands/add.py#L216

I think this is quite what you're looking for

**Comment by reyman** [comment]
Yep @alejandrogallo i found the function but i don't think this is that or i misunderstand something.

I run 

```
docs = papis.api.get_all_documents_in_lib(library_name)
# for each doc, rename folder
for doc in docs:
        new_name = doc["author_list"][0]["family"] + "." + doc["year"] + "." + doc["hash"]
```       

But at this step of program, calling this on existing library, i don't have the path or hash of the file linked with doc object/dict. I could use the add function and regenerate it, but that's better to parse the existing folder path or string nop ?
 

`
